src/m365/commands/login.spec.ts

```diff
@@ -296,6 +296,35 @@ describe(commands.LOGIN, () => {
     assert.strictEqual(auth.service.authType, AuthType.Browser, 'Incorrect authType set');
   });
 
+  it(`Don't start login flow when the CLI is signed in`, async () => {
+    sinon.stub(auth, 'ensureAccessToken').callsFake(() => Promise.resolve(''));
+
+    await command.action(logger, { options: { ensure: true } });
+    assert.strictEqual(auth.service.connected, true);
+  });
+
+  it(`Don't start login flow if the CLI is signed in as a user`, async () => {
+    sinon.stub(auth, 'ensureAccessToken').callsFake(() => Promise.resolve(''));
+
+    await command.action(logger, { options: { ensure: true, authType: 'password', userName: 'john.doe@contoso.com', password: 'password' } });
+    assert.strictEqual(auth.service.connected, true);
```

Comment:
Since we're not resetting the login state before each test, it will already be `true` even before we run this test, so this assets is not proving that the code is actually working.